app/views/movies/show.html.erb

```diff
@@ -0,0 +1,19 @@
+<h1><%= @movie.title %></h1>
+<p>Creation Year: <%= @movie.creation_year %></p>
+<p>Genre: <%= @movie.genre %></p>
+
+<h2>List of Actors:</h2>
+<ul>
+  <% @actors.each do |actor| %>
+    <li><%= actor.name %> (Age: <%= actor.age %>)</li>
+  <% end %>
+</ul>
+
+<p>Average Age: <%= @average_age %></p>
+
+<h2>Add Actor</h2>
+<%= form_with(url: "/movies/#{@movie.id}", method: :post) do |f| %>
```

Comment:
I disagree with your verb/route combination here... The user story says we should send in data for a movie, and an actor's id that isn't currently part of that movie, so that we can associate the two. In other words, we're _creating a relationship_ between those two instances. 

If we're creating something, we should use `post` as yo uhave here; but the path (url) informs us what is being _created_ as part of this request. If we go only on the URL, it says we're creating a movie by its ID (`/movies/:id`).... but that isn't quite right either, is it? We want to say what's being created in the URL. So, you have these two options: 
```
# 1
"/movies/:id/actors"
# 2
"/movie_actors"
```

With # 1, you have the movie ID in the route already. With # 2, you'd have to include a hidden field or a query param on the url indicating which movie ID to use.

Reply:
Overall, I would count this as 1 MVC infraction altogether.